asf_core_data/pipeline/mcs/test/test_process_mcs_installations.py

```diff
@@ -104,16 +87,21 @@ def test_regex_extraction_works():
 
 def test_same_lines_before_after():
     """
-    Checks if raw and processed datasets have the same number of lines.
+    Checks if raw and processed datasets have the same number of lines
+    (discounting the lines corresponding to test accounts and duplicate lines).
     """
-    assert len(raw_installations) == len(processed_installations)
+
+    assert len(
+        drop_instances_test_accounts(
+            raw_installations, "Installation Company Name"
+        ).drop_duplicates()
+    ) == len(processed_installations)
 
 
 def test_no_duplicate_lines():
     """
-    Checks that there are no duplicate lines in either raw or processed datasets
+    Checks that there are no duplicate lines in the processed dataset.
     """
-    assert len(raw_installations.drop_duplicates()) == len(raw_installations)
     assert len(processed_installations.drop_duplicates()) == len(
```

Comment:
If processed_installations is a pandas df, it might be better practice to compare the shape instead, either just the rows (.shape[0]) or even the entire shape. Do the number of columns have to match as well?

I've seen you do this on other places as well. Not an expert on tests but I haven't seen people compare Dataframes using length since that's not a very meaningul attribute if you can use shape instead.

```suggestion
    assert processed_installations.drop_duplicates().shape[0] == processed_installations.shape[0])
```

Reply:
hey @athrado,
- there's another test for the columns, it actually tests more than just the number of columns (tests the names);
- Regarding the use of `len()` vs. `shape[0]`, they do the exact same thing which is returning the number of lines. I don't find anything about using `shape[0]` being best practice instead of using `len()` - could you point me to any resource on this? I actually find `len()` more readable as people might mistake `shape[0]` by `shape[1]`. Here's a [stackoverflow post](https://stackoverflow.com/questions/15943769/how-do-i-get-the-row-count-of-a-pandas-dataframe) with multiple answers comparing `len()` and `shape`.

Reply:
@sofiapinto Fair enough, it could just be a matter of taste, that's why I said it _might_ be better practice. One of my tutors at uni critised us when using len() with numpy arrays of more than 1-dim, saying it should be avoided when working with tables because it could be unclear whether it's referring to rows or columns or any other dimension. And I always work with .shape myself so I guess I'm just more used to this notation. 

Do what feels right/better readable to you. In fact, len(df.index) might be a marginally faster, just looked it up.